src/cargo/sources/config.rs

```diff
@@ -207,7 +207,18 @@ restore the source replacement configuration to continue the build
         let mut srcs = Vec::new();
         if let Some(registry) = def.registry {
             let url = url(&registry, &format!("source.{}.registry", name))?;
-            srcs.push(SourceId::for_registry(&url)?);
+            if url.scheme().starts_with("rfc+") {
```

Comment:
We may want to think about what this prefix should be, or whether it shouldn't be a prefix at all and instead be a new key rather than reusing the `registry` key (like `local-registry` is).